linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -210,14 +210,15 @@ where
         // unrelated ones provided.
         let published_blob_ids = block.published_blob_ids();
         self.0
-            .check_no_missing_blobs(published_blob_ids.clone(), blobs)
+            .check_for_unneeded_blobs(&published_blob_ids, blobs)
             .await?;
         for blob in blobs {
             Self::check_blob_size(blob.content(), &policy)?;
             self.0.cache_recent_blob(Cow::Borrowed(blob)).await;
         }
+
         let checked_blobs = blobs.iter().map(|blob| blob.id()).collect::<BTreeSet<_>>();
-        for blob in self.0.get_blobs(published_blob_ids).await? {
+        for blob in self.0.get_cached_blobs(published_blob_ids).await? {
```

Comment:
If I understand correctly, we are relying on the fact that the blobs we just put into the cache are still there. That's very likely, but we shouldn't take it for granted, given that this is an LRU cache.

Let's make sure that if the user provided all blobs, this can't fail. I.e. let's only try to get the published blob IDs that aren't in `checked_blobs` already.

Alternatively, to keep things simple, we could just require for now that `blobs` must _always_ contain all published blobs? We can rethink that when we get to https://github.com/linera-io/linera-protocol/issues/2756.